### PR TITLE
Validation added in dynamic rows when pages changed

### LINKED ISSUE #28397: Magento UI Dynamic rows validate only the current page input elements not all the pages.
### Preconditions (*)

1. Magento 2.4

### Steps to reproduce (*)

1. Go to product edit page & click advanced pricing.
2. In customer group price dynamic rows, add 20 rows, fill all the inputs.
3. add one more row it will go to the next page, don't fill anything & then came back to the first page.
4. click done and save.

### Expected result (*)

1. It should validate all the inputs in the dynamic rows

### Actual result (*)

1. It does n't validate all the inputs.

image

Recorded case: https://github.com/magento/magento2/issues/28397#issuecomment-635495900

### PR DESCRIPTION
**Pull request by Mohamed-Asar**
### Description (*)
Dynamic rows validate only the current page input elements. it does n't validate all the page's inputs.
I Restricted the user to fill all the inputs before changing pages

### Fixed Issues (if relevant)

1. Fixes magento/magento2#28397

### Manual testing scenarios (*)

1. Go to product edit page & click advanced pricing.
2. In customer group price dynamic rows, add 20 rows, don't fill one or more required input fields
3. click add button it does n't allow to go next page

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by Mohamed-Asar** [comment]
@magento run all tests

**Comment by Mohamed-Asar** [comment]
@magento run Static Tests, Integration Tests, Functional Tests EE, Functional Tests B2B

File: app/code/Magento/Ui/view/base/web/js/dynamic-rows/dynamic-rows.js [hunk 1]
```diff
@@ -691,8 +693,13 @@ define([
             this.bubble('addChild', false);
 
             if (this.relatedData.length && this.relatedData.length % this.pageSize === 0) {
-                this.pages(this.pages() + 1);
-                this.nextPage();
+                this.validate();
+                if (this.valid) {
+                    this.pages(this.pages() + 1);
+                    this.nextPage();
+                } else {
```
Comment (VladimirZaets) [inline app/code/Magento/Ui/view/base/web/js/dynamic-rows/dynamic-rows.js]: You don't need `else` statement if you need just move out of the function. Please remove else statement.

File: app/code/Magento/Ui/view/base/web/js/dynamic-rows/dynamic-rows.js [hunk 2]
```diff
@@ -1145,6 +1158,37 @@ define([
...
+            };
+        },
+
+        /**
+         * Loop through all the inputs and validate them
+         *
+         * @return void
+         */
+        checkRows: function (elem) {
+            if (typeof elem === 'undefined') {
```
Comment (VladimirZaets) [inline app/code/Magento/Ui/view/base/web/js/dynamic-rows/dynamic-rows.js]: Please use underscore library to check the type of elements

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi @Mohamed-Asar. Thanks for collaboration. Please take a look at my review comments

**Comment by Mohamed-Asar** [comment]
@magento run all tests

**Review by sidolov — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-8125](https://jira.corp.magento.com/browse/ENGCOM-8125) has been created to process this Pull Request
:eight_spoked_asterisk: @sidolov, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi @Mohamed-Asar .
Could you cover the changes with auto tests?

Thanks!

**Comment by gabrieldagama** [comment]
The risk was set to`medium` due to: the suggested changes may affect other modules that use dynamic-rows.

**Comment by engcom-Golf** [comment]
Hi @Mohamed-Asar. 
Thank you for your collaboration.
Could you please address static build failures and consider covering expected behavior with an auto test?

**Comment by sidolov** [comment]
@Mohamed-Asar  I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**